tests/JustSaying.UnitTests/Messaging/Policies/CanCreatePolicy.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using JustSaying.Messaging.Middleware;
+using JustSaying.UnitTests.Messaging.Policies.ExamplePolicies;
+using Polly;
+using Polly.CircuitBreaker;
+using Shouldly;
+using Xunit;
+
+namespace JustSaying.UnitTests.Messaging.Policies
+{
+    public class CanCreatePolicy
+    {
+        [Fact]
+        public async Task PolicyBuilder_Async()
+        {
+            var called = false;
+            var noop = MiddlewareBuilder.BuildAsync<int, int>();
+
+            var result = await noop.RunAsync(1, async ct =>
+            {
+                called = true;
+                await Task.Delay(5, ct);
+                return 1;
+            }, CancellationToken.None);
+
+            called.ShouldBeTrue();
+            result.ShouldBe(1);
```

Comment:
It's not clear that the result is from the embedded lambda vs the first parameter to `RunAsync`.